lib/tbot/config/destination_directory.go

```diff
@@ -184,48 +237,154 @@ func (dd *DestinationDirectory) Verify(keys []string) error {
 		return nil
 	}
 
+	errors := []error{}
+	for _, key := range keys {
+		path := filepath.Join(dd.Path, key)
+
+		errors = append(errors, botfs.VerifyLegacyACL(path, &botfs.ACLOptions{
+			BotUser: currentUser,
+		}))
+	}
+
+	// Unlike new-style ACLs, we'll allow hard fails here: we don't expect to
+	// be able to manage legacy ACLs at runtime and need to depend on them being
+	// correct from the start, so hard fails at this stage make sense.
+	aggregate := trace.NewAggregate(errors...)
+	if dd.ACLs == botfs.ACLRequired {
+		// Hard fail if ACLs are specifically requested and there are errors.
+		return aggregate
+	}
+
+	if aggregate != nil {
+		log.WarnContext(
+			context.TODO(),
+			"Destination has unexpected ACLs",
+			"path", dd.Path,
+			"errors", aggregate,
+		)
+	}
+
+	return nil
+}
+
+func (dd *DestinationDirectory) Verify(keys []string) error {
```

Comment:
These checks were largely rewritten but maintain behavior for legacy ACLs. In retrospect, I'm not sure how much of this ever belonged in `Verify()` - which runs before every write - and instead may have made more sense in `Init()` - which runs once at startup. It bails out before doing any proper checks for new-style ACLs, since those are actually checked at the start of `Write()`.